src/amber/router/pipe/csrf.cr

```diff
@@ -66,8 +66,8 @@ module Amber
 
         def valid_token?(context)
           if request_token(context) && real_session_token(context)
-            decoded_request = Base64.decode(request_token(context).to_s)
-            return false unless decoded_request.size == TOKEN_LENGTH * 2
+            decoded_request = Base64.decode(request_token(context).to_s) rescue nil
```

Comment:
My only concern is that this adds another check that gets called on every request.

I'd recommend returning on rescue instead of checking for nil.

Reply:
```cr
def valid_token?(context)
  if request_token(context) && real_session_token(context)
    decoded_request = Base64.decode(request_token(context).to_s) rescue return false
    return false if decoded_request.nil? || decoded_request.size != TOKEN_LENGTH * 2 

    unmasked = TokenOperations.unmask(decoded_request)
    session_token = Base64.decode(real_session_token(context))
    return Crypto::Subtle.constant_time_compare(unmasked, session_token)
  end 
  return false
end
```

Or better yet...

```cr
def valid_token?(context)
  if request_token(context) && real_session_token(context)
    decoded_request = Base64.decode(request_token(context).to_s)
    return false if decoded_request.nil? || decoded_request.size != TOKEN_LENGTH * 2 

    unmasked = TokenOperations.unmask(decoded_request)
    session_token = Base64.decode(real_session_token(context))
    return Crypto::Subtle.constant_time_compare(unmasked, session_token)
  end
  return false
rescue e : Base64::Error
  return false
end
```